### ISSUE TITLE
check_dig invokes dig with arguments in wrong order

### ISSUE THREAD

**Issue by hansmi** [nagios-plugins/nagios-plugins#233]
We observed this when [EPEL](https://fedoraproject.org/wiki/EPEL) updated their nagios-plugins package from version 2.0.3 to 2.1.4 on approximately January 10, 2017. Our monitoring system invokes `check_dig` to look up `version.bind`:

```
check_dig -l version.bind -T TXT -A '-c chaos'
```

The expected output is like it was in version 2.0.3, i.e. `DNS OK …`. Instead we get this:

```
$ check_dig -l version.bind -T TXT -A '-c chaos'
DNS CRITICAL - 0.037 seconds response time (No ANSWER SECTION found)|time=0.036510s;;;0.000000
```

`dig` is invoked as `/usr/bin/dig -c chaos  -p 53 @127.0.0.1 version.bind TXT +tries=3 +time=4` (retrieved using `-v`). Executing a query for an `A` record (`IN` class) clearly shows that `dig` misunderstands the arguments:

```
$ check_dig -v -l heise.de -T TXT -A '-c IN'
/usr/bin/dig -c IN  -p 53 @127.0.0.1 heise.de TXT +tries=3 +time=4
Looking for: 'heise.de'
;; ANSWER SECTION:
heise.de.		2102	IN	A	193.99.144.80
DNS OK - 0.030 seconds response time (heise.de.  2102 IN A 193.99.144.80)|time=0.029852s;;;0.000000

$ /usr/bin/dig -c IN  -p 53 @127.0.0.1 heise.de TXT +tries=3 +time=4
; <<>> DiG 9.9.4-RedHat-9.9.4-38.el7_3.1 <<>> -c IN -p 53 @127.0.0.1 heise.de TXT +tries=3 +time=4
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 8690
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;heise.de.			IN	A

;; ANSWER SECTION:
heise.de.		2079	IN	A	193.99.144.80

;; Query time: 6 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Die Jan 31 18:32:37 CET 2017
;; MSG SIZE  rcvd: 42

;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 35030
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;TXT.				IN	A

;; Query time: 2 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Die Jan 31 18:32:37 CET 2017
;; MSG SIZE  rcvd: 21
```

There are two answer sections, the first for `heise.de.`, the second for `TXT.`, because the order of arguments upon calling `dig`  is wrong. Commit 9a70f22b74d in December 2013 changed the order of arguments and is included in version 2.1.4 (and a few after 2.0.3; I didn't look up the exact versions).

**Comment by jfrickson** [comment]
Looking into it...

**Comment by jfrickson** [comment]
Retrieval of TXT records was messed up, but has been fixed in version 2.2.1 which is due to be released on April 19.

You can verify by using branch `release-2.2.1-rc1` at https://github.com/nagios-plugins/nagios-plugins/archive/release-2.2.1-rc1.tar.gz
